rectools/visuals/visual_app.py

```diff
@@ -0,0 +1,273 @@
+    ...     formatters=formatters,
+    ...     auto_display=False
+    ... )
+    """
+
+    def __init__(
+        self,
+        recos: tp.Dict[tp.Hashable, pd.DataFrame],
+        item_data: pd.DataFrame,
+        requests_dict: tp.Dict[tp.Hashable, tp.Hashable],
+        is_u2i: bool = True,
+        interactions: tp.Optional[pd.DataFrame] = None,
+        auto_display: bool = True,
+        formatters: tp.Optional[tp.Dict[str, tp.Callable]] = None,
+        rows_limit: int = 20,
+    ) -> None:
+        self.data_storage = AppDataStorage(
+            interactions=interactions,
+            recos=recos,
+            requests_dict=requests_dict,
+            item_data=item_data,
+            is_u2i=is_u2i,
+        )
+        self.rows_limit = rows_limit
+        self.formatters = formatters if formatters is not None else {}
+        if auto_display:
+            self.display()
+
+    def _convert_to_html(self, df: pd.DataFrame) -> str:
+        html_repr = (
+            df.to_html(
+                escape=False,
+                index=False,
+                formatters=self.formatters,
+                max_rows=self.rows_limit,
+                border=0,
+            )
+            .replace("<td>", '<td align="center">')
+            .replace("<th>", '<th style="text-align: center; min-width: 100px;">')
+        )
+        return html_repr
+
+    def _display_interactions(self, request_name: str) -> None:
+        """Display viewed items for `request_name`"""
+        items_tab = widgets.Tab()
+        df = self.data_storage.processed_interactions[request_name]
+        items_tab.children = [widgets.HTML(value=self._convert_to_html(df))]
+        items_tab.set_title(index=0, title="Interactions")
+        display(items_tab)
+
+    def _display_recos(self, request_name: str, model_name: str) -> None:
+        """Display recommended items for `request_name` from model `model_name`"""
+        items_tab = widgets.Tab()
+        df = self.data_storage.processed_recos[model_name][request_name]
+        items_tab.children = [widgets.HTML(value=self._convert_to_html(df))]
+        items_tab.set_title(index=0, title="Recos")
+        display(items_tab)
+
+    def _display_request_id(self, request_name: str) -> None:
+        """Display request_id for `request_name`"""
+        request_id = self.data_storage.requests_dict[request_name]
+        display(widgets.HTML(value=f"{self.data_storage.request_colname}: {request_id}"))
+
+    def _display_model_name(self, model_name: str) -> None:
+        """Display model_name"""
+        display(widgets.HTML(value=f"Model name: {model_name}"))
+
+    def display(self) -> None:
+        """Display full VisualApp widget"""
+        request_name_selection = widgets.ToggleButtons(
+            options=self.data_storage.request_names,
+            description=f"Request {self.data_storage.request_colname}:",
+            disabled=False,
+            button_style="warning",
```

Comment:
I understand that you just need yellow color, but probably there is way to apply it without using warning style?

Reply:
I wish there was a better way. But it is only possible through some hacks like these:
- https://stackoverflow.com/questions/68643117/python-ipywidgets-togglebutton-style-color
- https://stackoverflow.com/questions/72504234/styling-ipywidgets-accordion-panels-individually

Until ipywidgets add support for `button_color` property in `ToggleButtons.style` we are stuck with predefined colors

Reply:
Hacks don't look too bad

Let's keep it as is, but add some comments please with this links probably

Reply:
They are hard to support and impossible to check during linters and tests. So I really don't want to add them. I've added explicit comment to code.